libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -160,6 +163,31 @@ void AP_DDS_Client::populate_static_transforms(tf2_msgs_msg_TFMessage& msg)
 
 }
 
+void AP_DDS_Client::update_topic(geometry_msgs_msg_PoseStamped& msg)
+{
+    update_topic(msg.header.stamp);
+    strcpy(msg.header.frame_id, ENU_FRAME_ID);
```

Comment:
Frame name should be "base_link", which is already set as `BASE_LINK_FRAME_ID` 
https://automaticaddison.com/coordinate-frames-and-transforms-for-ros-based-mobile-robots/

Reply:
Done! 
![Screenshot from 2023-04-14 01-03-29](https://user-images.githubusercontent.com/62964137/231939333-e2b31118-2293-460d-971f-61f40833abad.png)

Reply:
This section of the REP applies. 
```
The transform from odom to base_link is computed and broadcast by one of the odometry sources.

The transform from map to base_link is computed by a localization component. However, the localization component does not broadcast the transform from map to base_link. Instead, it first receives the transform from odom to base_link, and uses this information to broadcast the transform from map to odom.
```
https://answers.ros.org/question/50870/what-frame-is-sensor_msgsimuorientation-relative-to/
^ More insight on frames

https://answers.ros.org/question/210124/which-frame_id-do-i-usually-specify-for-the-navsatfix-header/
^ NavSatFix

Give me some time to put together a diagram and we can go over it.

Reply:
> You should put in the frame_id in which the sensor is mounted (the transform from base_link).

I got it now, that makes sense